storage/remote/queue_manager.go

```diff
@@ -661,9 +656,9 @@ func (s *shards) runShard(ctx context.Context, i int, queue chan prompb.TimeSeri
 	// Send batches of at most MaxSamplesPerSend samples to the remote storage.
 	// If we have fewer samples than that, flush them out after a deadline
 	// anyways.
-	pendingSamples := []prompb.TimeSeries{}
-
 	max := s.qm.cfg.MaxSamplesPerSend
+	pendingSamples := make([]prompb.TimeSeries, 0, max)
```

Comment:
I think this has the potential to cause more backup problems.

Reply:
Could you explain this comment more? I am not seeing how this would cause backup problems.

Reply:
Sorry, I was thinking this would enforce there being no more than `maxSamplesPerSend` pending samples, but that's not how slices work :man_facepalming:  

Backup meaning the shards not being able to send samples at the rate their being scrapped and written to the WAL. At the moment we send as soon as we reach `maxSamplesPerSend` or the flush timeout. At the moment we can potentially have some multiple of `maxSamplesPerSend` samples pending, meaning that after a successful send there's theoretically less time before we have enough samples for the next send. If we enforced only have `maxSamplesPerSend` pending it could take longer to fill up `pendingSample` given that we block on the shards queue.